### PR TITLE
[GEOT-7556] Generate javadoc for Maven deploy

### PR DESCRIPTION
**Pull request by roarbra**
[GEOT-7556](https://osgeo-org.atlassian.net/browse/GEOT-7556) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

There is no Javadoc jars published alongside the jar and sources. I believe this is because of a missing javadoc:jar call before the install.

In addition I've removed a warning for wrong configuration (additionalparam), and the keywords option is just necessary for the javadoc that are published to a web site.

# Checklist

- [X] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [X] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [X] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [X] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [X] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [X] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [X] Bug fixes and small new features are presented as a single commit.
- [X] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: build/release/build_release.sh [hunk 1]
```diff
@@ -142,14 +143,14 @@ popd > /dev/null
 if [ "$SKIP_BUILD" != true ]; then
   export MAVEN_OPTS="-Xmx2048m"
   echo "building release"
-  mvn $MAVEN_FLAGS -DskipTests -Dall clean install
+  mvn $MAVEN_FLAGS -DskipTests -Dall clean generate-sources javadoc:jar install
```
Comment (aaime) [inline build/release/build_release.sh:146]: I'm wondering if this helps at all, given that release publish does this:

```
mvn clean deploy -DskipTests -Dall
```

See that "clean"? I'm guessing that if you want to deploy javadocs, the building of them should be done there?
Also, the sources are already packaged, don't think we need to bother the build_release script with them, e.g.:

image

@jodygarnett knows the release procedures much better than I do, and might provide some more suggestions.

**Comment by mprins** [comment]
probably you want something like

```xml

            <plugin>
                <artifactId>maven-javadoc-plugin</artifactId>
                <executions>
                    <execution>
                        <id>attach-javadoc</id>
                        <goals>
                            <goal>jar</goal>
                        </goals>
                        <phase>package</phase>
   ...

```
to "attach" the javadoc jar files as artifacts (possibly using `verify` phase instead)

**Comment by roarbra** [comment]
> probably you want something like
> 
> ```
>             <plugin>
>                 <artifactId>maven-javadoc-plugin</artifactId>
>                 <executions>
>                     <execution>
>                         <id>attach-javadoc</id>
>                         <goals>
>                             <goal>jar</goal>
>                         </goals>
>                         <phase>package</phase>
>    ...
> ```
> 
> to "attach" the javadoc jar files as artifacts (possibly using `verify` phase instead)

I'm afraid that would make them build each time users are building with Maven. Don't think everyone would be happy about that. Having an exclusive goal is fine by me, as long as it's called within the release cycle.

File: build/release/build_release.sh (hunk 1 above)
Comment (roarbra) [reply build/release/build_release.sh:146]: > I'm wondering if this helps at all, given that release publish does this:
> 
> ```
> mvn clean deploy -DskipTests -Dall
> ```

It was somewhat hard to figure out how the release process worked. You're absolutely correct that it was unnecessary to add that generate-sources. Not sure why I did that.

Where in the scripts is this mvn-call you are referring to?

Comment (roarbra) [reply build/release/build_release.sh:146]: > 
> It was somewhat hard to figure out how the release process worked. You're absolutely correct that it was unnecessary to add that generate-sources. Not sure why I did that.
> 

I remember now. And it is actually necessary. It isn't the sources.jar that is generated, but the Java-sources we generate from xsd and such. Javadoc-generation will fail without that goal.

Comment (jodygarnett) [reply build/release/build_release.sh]: Can we add comment so we do not have to remember this detail again in the future?

**Pull request merged**